packages/core/src/runtime/useEventBus.ts

```diff
@@ -1,7 +1,8 @@
+import type { EventBus, ValidEventTypes } from "../index.ts";
 import { useRuntime } from "./RuntimeContext.ts";
 
-export function useEventBus() {
+export function useEventBus<EventTypes extends ValidEventTypes>() {
     const runtime = useRuntime();
 
-    return runtime.eventBus;
+    return runtime.eventBus as unknown as EventBus<EventTypes>;
```

Comment:
`runtime.eventBus as unknown as EventBus<EventTypes` is ugly but it's the only way to type classes with different generics into each others